### PR TITLE
Migrate postman to jest (first part)

### PR DESCRIPTION
**Pull request by gcusnieux**
**Issue**

https://github.com/gravitee-io/issues/issues/7587

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/7587-migrate-postman-run-e2e/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-e2e/api-test/apis/api-definition.spec.ts [hunk 1]
```diff
@@ -749,9 +749,12 @@ afterAll(async () => {
...
   expect(apiStoppedResponse.raw.status).toEqual(204);
   const apiPagesApi = new APIPagesApi(configuration);
+  const deleteApiPages = [];
   for (const pageId of pageIds.reverse()) {
-    await apiPagesApi.deleteApiPage({ orgId, envId, api, page: pageId });
+    deleteApiPages.push(apiPagesApi.deleteApiPage({ orgId, envId, api, page: pageId }));
   }
 
-  await apisApi.deleteApi({ orgId, envId, api: api });
+  await Promise.all(deleteApiPages);
```
Comment (gaetanmaisse) [inline gravitee-apim-e2e/api-test/apis/api-definition.spec.ts]: Is it ok to delete everything in parallel? If so then we can simplify a bit L752:

```ts
  const deleteApiPages = pageIds.map(page => apiPagesApi.deleteApiPage({ orgId, envId, api, page }))

  await Promise.all(deleteApiPages);
```

File: gravitee-apim-e2e/api-test/apis/api-search.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      orgId,
+      envId,
+      newApiEntity: ApisFaker.newApi({ name: ApisFaker.uniqueWord() }),
+    }),
+  ]);
+  return [userApi, adminApi];
+});
+
+describe('API Search', () => {
+  test('should found API of API_USER as ADMIN_USER', async () => {
```
Comment (gaetanmaisse) [inline gravitee-apim-e2e/api-test/apis/api-search.spec.ts]:
```suggestion
  test('should find API of API_USER as ADMIN_USER', async () => {
```

Comment (gaetanmaisse) [reply gravitee-apim-e2e/api-test/apis/api-search.spec.ts]: Same for the other ⬇️

File: gravitee-apim-e2e/api-test/apis/api-search.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    const apiListItems = await apisResourceAdmin.searchApis({ orgId, envId, q: userApi.name });
+    expect(apiListItems.length).toEqual(1);
```
Comment (gaetanmaisse) [inline gravitee-apim-e2e/api-test/apis/api-search.spec.ts]:
```suggestion
    expect(apiListItems).toHaveLength(1);
```

Same for the other ⬇️

File: gravitee-apim-e2e/api-test/apis/api-publishing.spec.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    envId,
+    newApiEntity: ApisFaker.newApi(),
+  });
+  return userApi;
+});
+
+describe('Portal: API publishing', () => {
+  test('should get APIs without unpublished as Anonymous', async () => {
+    const apisResponse = await apiResource.getApis({});
+    expect(apisResponse.data.find((api) => api.id === userApi.id)).not.toBeDefined();
```
Comment (gaetanmaisse) [inline gravitee-apim-e2e/api-test/apis/api-publishing.spec.ts]: `apisResponse.data` is empty here so it feels weird to do a find, maybe we should add other data and check for the length too

Comment (gcusnieux) [reply gravitee-apim-e2e/api-test/apis/api-publishing.spec.ts]: Agree but the test case is not complete, I just copied this one to validate the configuration and the client portal. 
I prefer to improve the tests after the migration.

**Review by gaetanmaisse — APPROVED** [review]

**Pull request merged**